core/manifest/ThemeManifest.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+/**
+ * A class which builds a manifest of all themes (which is really just a directory called "templates")
+ *
+ * @package framework
+ * @subpackage manifest
+ */
+class ThemeManifest {
```

Comment:
Maybe as well introduce it within a namespace from the start? We did this, e.g., with new assets work.

Arguably this is more a part of "view" than "core"—if you were to remove the view you'd also remove this.